components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AbstractAuthenticatorAdapter.java

```diff
@@ -62,49 +66,52 @@ public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletRe
                                            AuthenticationContext context)
             throws AuthenticationFailedException, LogoutFailedException {
 
-        if (context.isLogoutRequest()) {
-            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
-        }
+        try {
+            if (context.isLogoutRequest()) {
+                return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+            }
 
-        /* TODO: Catch AuthenticationFailedException error and continue with super.process, and handle error at
-            processAuthenticationResponse method. */
-        context.removeProperty(AuthenticatorAdapterConstants.EXECUTION_STATUS_PROP_NAME);
-        Map<String, Object> eventContext = new HashMap<>();
-        eventContext.put(AuthenticatorAdapterConstants.AUTH_REQUEST, request);
-        eventContext.put(AuthenticatorAdapterConstants.AUTH_RESPONSE, response);
-        eventContext.put(AuthenticatorAdapterConstants.AUTH_CONTEXT, context);
-        ActionExecutionStatus executionStatus = executeAction(context, eventContext, context.getTenantDomain());
-        context.setProperty(AuthenticatorAdapterConstants.EXECUTION_STATUS_PROP_NAME, executionStatus);
-
-        if (executionStatus.getStatus() == ActionExecutionStatus.Status.INCOMPLETE) {
-            context.setCurrentAuthenticator(getName());
-            context.setRetrying(false);
-            return AuthenticatorFlowStatus.INCOMPLETE;
-        }
+            Map<String, Object> eventContext = new HashMap<>();
+            eventContext.put(AuthenticatorAdapterConstants.AUTH_REQUEST, request);
+            eventContext.put(AuthenticatorAdapterConstants.AUTH_RESPONSE, response);
+            eventContext.put(AuthenticatorAdapterConstants.AUTH_CONTEXT, context);
+            eventContext.put(AuthenticatorAdapterConstants.AUTH_TYPE, getAuthenticationType());
 
-        return super.process(request, response, context);
+            ActionExecutionStatus executionStatus = executeAction(context, eventContext, context.getTenantDomain());
+
+            if (executionStatus.getStatus() == ActionExecutionStatus.Status.INCOMPLETE) {
+                context.setCurrentAuthenticator(getName());
+                context.setRetrying(false);
+                return AuthenticatorFlowStatus.INCOMPLETE;
+            }
+
+            return super.process(request, response, context);
+        } catch (ActionExecutionException e) {
+            context.setProperty(FrameworkConstants.AUTH_ERROR_CODE, e.getMessage());
+            context.setProperty(FrameworkConstants.AUTH_ERROR_MSG, "An error occurred while authenticating with " +
+                    "user the external authentication authentication service.");
+            return super.process(request, response, context);
+        } finally {
+            context.removeProperty(AuthenticatorAdapterConstants.EXECUTION_STATUS_PROP_NAME);
+        }
```

Comment:
Is this really needed ?

Reply:
As the this property only used within the scope of process method and must set new status at each process method invocation.